### PR TITLE
Cover some Mobile notifications by Unit Tests

### PR DESCRIPTION
**Pull request by LevchenkoS**
Has been covered:
- `OnAppInterfaceUnregisteredNotification`;
- `OnAudioPassThruNotification`;
- `OnLanguageChangeNotification`;
- `OnPermissionsChangeNotification`;
- `OnDriverDistractionNotification`.

Removed redundant code from
`OnPermissionsChangeNotification::Run()`
Reason: copies part of `SendNotification()` method.

Related to: [APPLINK-26532](https://adc.luxoft.com/jira/browse/APPLINK-26532)

@AByzhynar, @okozlovlux, @Kozoriz, @LuxoftAKutsan, @vlantonov, @wolfylambova22, please review.

### CODE REVIEW

**Comment by okozlovlux** [comment]
@LevchenkoS reviewed

File: src/components/application_manager/test/commands/mobile/simple_notification_commands_test.cc [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+         (*arg)[am::strings::params][am::strings::message_type].asInt() ==
+             application_manager::MessageType::kNotification;
+}
+
+TYPED_TEST_CASE(MobileNotificationCommandsTest, NotificationCommandsList);
+
+TYPED_TEST(MobileNotificationCommandsTest, Run_SendMessageToMobile_SUCCESS) {
+  ::utils::SharedPtr<typename TestFixture::CommandType> command =
+      this->template CreateCommand<typename TestFixture::CommandType>();
+  EXPECT_CALL(this->app_mngr_,
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/simple_notification_commands_test.cc]: @LevchenkoS as I understand this here is redundant, no?

Comment (LevchenkoS) [inline src/components/application_manager/test/commands/mobile/simple_notification_commands_test.cc]: @LuxoftAKutsan, It's required by templates.

File: src/components/application_manager/src/commands/mobile/on_permissions_change_notification.cc [hunk 2]
```diff
@@ -48,9 +48,6 @@ OnPermissionsChangeNotification::~OnPermissionsChangeNotification() {}
 void OnPermissionsChangeNotification::Run() {
   SDL_AUTO_TRACE();
 
-  (*message_)[strings::params][strings::message_type] =
-      static_cast<int32_t>(application_manager::MessageType::kNotification);
-
```
Comment (AByzhynar) [inline src/components/application_manager/src/commands/mobile/on_permissions_change_notification.cc]: @LevchenkoS Why did you delete this code within the task of UT covering?

File: src/components/application_manager/test/commands/mobile/simple_notification_commands_test.cc (hunk 1 above)
Comment (AByzhynar) [inline src/components/application_manager/test/commands/mobile/simple_notification_commands_test.cc]: @LevchenkoS It's not the first nor the last time you were asked about this. Add comment explaining why `this` is needed

File: src/components/application_manager/src/commands/mobile/on_permissions_change_notification.cc (hunk 2 above)
Comment (LevchenkoS) [inline src/components/application_manager/src/commands/mobile/on_permissions_change_notification.cc]: @AByzhynar, this lines are redundant

**Comment by vlantonov** [comment]
@LevchenkoS Reviewed

**Comment by wolfylambova22** [comment]
Reviewed.

**Comment by Kozoriz** [comment]
Reviewed

**Comment by LuxoftAKutsan** [comment]
Reviwed
